src/Host/Broker/Impl/RemoteUri/RemoteUriHelper.cs

```diff
@@ -1,78 +1,154 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-
 using System;
 using System.Collections.Generic;
-using System.Text;
-using System.Threading.Tasks;
-using System.Net.Http;
 using System.IO;
-using Newtonsoft.Json;
+using System.Net;
+using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
-using System.Net.Http.Headers;
+using Microsoft.Common.Core;
 
 namespace Microsoft.R.Host.Broker.RemoteUri {
     public class RemoteUriHelper {
         public static async Task HandlerAsync(HttpContext context) {
-            BinaryReader reader = new BinaryReader(context.Request.Body);
-            var url = reader.ReadString();
-            var method = reader.ReadString();
-            Dictionary<string, string> headers = JsonConvert.DeserializeObject<Dictionary<string, string>>(reader.ReadString());
+            var url = context.Request.Headers["x-rtvs-url"];
+
+            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(url);
+            request.Method = context.Request.Method;
+            request.UseDefaultCredentials = true;
+            SetRequestHeaders(request, context.Request.Headers);
+
+            if (context.Request.ContentLength > 0) {
+                Stream reqStream = await request.GetRequestStreamAsync();
+                await context.Request.Body.CopyToAsync(reqStream);
+                await reqStream.FlushAsync();
+            }
 
-            HttpClient client = new HttpClient();
-            Uri uri = new Uri(url);
+            HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync();
 
-            HttpMethod httpMethod = new HttpMethod(method);
-            HttpRequestMessage req = new HttpRequestMessage(httpMethod, uri);
+            SetResponseHeaders(response, context.Response);
 
-            foreach (var key in headers.Keys) {
-                var val = headers[key];
-                req.Headers.Add(key, val);
+            Stream respStream = response.GetResponseStream();
+            await respStream.CopyToAsync(context.Response.Body);
+            await context.Response.Body.FlushAsync();
+        }
+
+        private static long GetLong(string value) {
+            long ret;
+            if (long.TryParse(value, out ret)) {
```

Comment:
`TryParse` will write the default value into the out-parameter if it fails to parse, so you can just `return ret` unconditionally.